crates/runtime/src/dataconnector/mysql.rs

```diff
@@ -80,12 +80,40 @@ impl DataConnectorFactory for MySQLFactory {
                 dyn DbConnectionPool<mysql_async::Conn, &'static (dyn ToValue + Sync)>
                     + Send
                     + Sync,
-            > = Arc::new(
-                MySQLConnectionPool::new(params.to_secret_map())
-                    .await
-                    .context(UnableToCreateMySQLConnectionPoolSnafu)?,
-            );
-
+            > = match MySQLConnectionPool::new(params.to_secret_map()).await {
+                Ok(pool) => Arc::new(pool),
+                Err(error) => match error {
+                    mysqlpool::Error::InvalidUsernameOrPassword { .. } => {
```

Comment:
If mysqlpool::Error::InvalidUsernameOrPassword is not always the same and contain details I would recommend to keep it for troubleshooting reason, otherwise it could be hard to identify the problem. For example if env key is configured incorrectly resulting in empty password

Reply:
Can be updated next IMO

Reply:
Yeah let me verify whether different wrong configuration of username / password would result in different error messages. IIRC the error messages are the same for wrong password / empty password / wrong username, etc.